### PR TITLE
Create foreign-event IQ handler and verify requests

### PR DESCRIPTION
**Pull request by arkgil**
This PR installs handler for foreign-event IQ stanzas and simple verification of requests format.

### CODE REVIEW

File: apps/ejabberd/src/foreign_event/mod_foreign.erl [hunk 1]
```diff
@@ -25,6 +25,9 @@
```
Comment (arcusfelis) [inline apps/ejabberd/src/foreign_event/mod_foreign.erl]: module description is missing.

File: apps/ejabberd/src/foreign_event/mod_foreign.erl [hunk 2]
```diff
@@ -73,6 +79,20 @@ get_disco_items(empty, From, To, Node, Lang) ->
...
     Acc.
 
+-spec iq_handler(From :: jid(), To :: jid(), iq()) -> iq() | ignore.
+iq_handler(_From, _To, #iq{type = get} = IQ) ->
+    IQ#iq{type = error, sub_el = ?ERR_NOT_ALLOWED};
+iq_handler(_From, _To, #iq{type = set, sub_el = ForeignEvent} = IQ) ->
+    case parse_foreign_event(ForeignEvent) of
+        {ok, Type, Request, _Publish} ->
+            maybe_dispatch_request(Type, Request, IQ);
+        error ->
```
Comment (arcusfelis) [inline apps/ejabberd/src/foreign_event/mod_foreign.erl]: {error, missing_request_element} | {error, missing_type_field} is more preferable.

File: apps/ejabberd/src/foreign_event/mod_foreign_http.erl [hunk 3]
```diff
@@ -28,9 +28,11 @@
 %%--------------------------------------------------------------------
 
 make_request(Request, OnResponse) ->
-    {Host, Path, Method, Headers, Body} = parse(Request),
-    Response = do_make_request(Host, Path, Method, Headers, Body),
-    respond(encode(Response), OnResponse),
+    spawn(fun() ->
```
Comment (arcusfelis) [inline apps/ejabberd/src/foreign_event/mod_foreign_http.erl]: I would like to receive IQ result after I've made a request.
Or an IQ error if the server cannot contact http server or any other problem.

File: apps/ejabberd/src/foreign_event/mod_foreign.erl (hunk 2 above)
Comment (mentels) [inline apps/ejabberd/src/foreign_event/mod_foreign.erl]: @arcusfelis what exactly do you mean?

File: apps/ejabberd/src/foreign_event/mod_foreign_http.erl (hunk 3 above)
Comment (mentels) [inline apps/ejabberd/src/foreign_event/mod_foreign_http.erl]: @arcusfelis the use case we have is that we need to make the requests asynchronously. The actual response of the HTTP request is to be published to MUC or PubSub.

Comment (rslota) [inline apps/ejabberd/src/foreign_event/mod_foreign_http.erl]: If you need asynchronous request, there is one thing I need to point out: You really shouldn't do that using `spawn` directly. This way you just cannot control the number of process in the system what may be dangerous or at least "not fair" to other modules. There should be a pool for that.

Comment (arkgil) [inline apps/ejabberd/src/foreign_event/mod_foreign_http.erl]: There will be a pool for that :) This PR introduced only IQ handler, `spawn` is just a placeholder. There will be aconfigurable pool later :)

**Pull request merged**